libraries/common/char.effekt

```diff
@@ -15,32 +15,30 @@ def isWhitespace(c: Char): Bool = c match {
   case _      => false
 }
 
+/// Gets the value of a given ASCII digit in base 10
+def digitValue(c: Char): Option[Int] =
+  if (c >= '0' && c <= '9') {
+    Some(c.toInt - '0'.toInt)
+  } else {
+    None()
+  }
+
+/// Gets the value of a given ASCII digit in base 16
+def hexDigitValue(c: Char): Option[Int] =
+  c match {
+    case c and c >= '0' && c <= '9' => Some(c.toInt - '0'.toInt)
+    case c and c >= 'A' && c <= 'F' => Some((c.toInt - 'A'.toInt) + 10)
+    case c and c >= 'a' && c <= 'f' => Some((c.toInt - 'a'.toInt) + 10)
+    case _ => None()
+  }
+
 /// Checks if the given character is an ASCII digit in base 10
 /// Use `digitValue(c: Char)` to get the numeric value out.
-def isDigit(c: Char): Bool = { c >= '0' && c <= '9' }
-
-/// Checks if the given character is an ASCII digit in the given base
-/// Use `digitValue(c: Char, base: Int)` to get the numeric value out.
-def isDigit(c: Char, base: Int): Bool = {
-  with on[WrongFormat].default { false }
-  val _ = digitValue(c, base)
-  true
-}
+def isDigit(c: Char): Bool = digitValue(c).isDefined
```

Comment:
The tests are still using the `isDigit(char: Char, base: Int)` version. 👀
I think it's kind of a shame that we're getting rid of the versions with an explicit base for seemingly no reason, but I'll be able to live with it...